// Copyright 2023 Google LLC
//
// Use of this source code is governed by an MIT-style license that can be found
// in the LICENSE file or at https://opensource.org/licenses/MIT.
syntax = "proto3";

package rrg.action.get_file_contents;

import "grr_response_proto/rrg/fs.proto";

message Args {
  // Absolute path to the file to get the contents of.
  rrg.fs.Path path = 1;

  // Byte offset from which the content should be retrieved.
  //
  // If unset, starts from the beginning of the file.
  uint64 offset = 2;

  // Number of bytes to from the file from the given offset to fetch.
  //
  // If unset, collects the entire file (possibly in multible results).
  uint64 length = 3;
}

message Result {
  // A byte offset of the file part sent to the blob sink.
  uint64 offset = 1;

  // A number of bytes of the file part sent to the blob sink.
  uint64 length = 2;

  // A SHA-256 hash of the file part sent to the blob sink.
  bytes blob_sha256 = 3;
}
